tiledb/sm/filter/filter.h

```diff
@@ -79,16 +80,17 @@ class Filter {
    * @param type Filter type to create
    * @return New Filter instance or nullptr on error.
    */
-  static Filter* create(FilterType type);
+  static std::tuple<Status, std::optional<std::shared_ptr<Filter>>> create(
+      FilterType type);
 
   /**
    * Deserializes a new Filter instance from the data in the given buffer.
    *
    * @param buff The buffer to deserialize from.
-   * @param filter New filter instance (caller's responsibility to free).
-   * @return Status
+   * @return Status and Filter pointer
    */
-  static Status deserialize(ConstBuffer* buff, Filter** filter);
+  static std::tuple<Status, std::optional<std::shared_ptr<Filter>>> deserialize(
+      ConstBuffer* buff);
```

Comment:
Prefer a `const T&` when possible over `T*`.

If there were a large number of calls to this function, it might be worth changing the function signature in a separate PR. `deserialize`, though, is only called in a handful of places.

Reply:
ConstBuffer is derived from BufferBase. read and write functions are defined in base class BufferBase. It is more easy for us to call those functions(read, write, and etc) if we use pointer.  If we use const ConstBuffer&, we need add override functions in ConstBuffer class. so it is better to do it in a separate PR.